makefile

```diff
@@ -1,29 +1,134 @@
-	@echo "PROJECT: $(PROJECT)"
-	@echo "REGION: $(REGION)"
-	@echo "PROFILE: $(PROFILE)"
-	@echo "COMMIT_SHA: $(COMMIT_SHA)"
-	@echo "IMAGE_TAG: $(IMAGE_TAG)"
-	@echo "VERSION_LABEL: $(VERSION_LABEL)"
-
-# Local Development
+export REPO_LOCATION=$(shell git rev-parse --show-toplevel)
+
+define TFVARS_DATA
+project_name = "$(PROJECT_NAME)"
+target_env = "$(ENV_NAME)"
+project_code = "$(PROJECT_CODE)"
+git_version = "ETS-$(COMMIT_SHA)"
+api_zip = "build/server.zip"
+ets_to_sbc_zip = "build/etsToSbc.zip"
+phac_to_sbc_zip = "build/phacToSbc.zip"
+endef
+export TFVARS_DATA
+
+define TF_BACKEND_CFG
+workspaces { name = "$(LZ2_PROJECT)-$(ENV_NAME)" }
+hostname     = "app.terraform.io"
+organization = "bcgov"
+endef
+export TF_BACKEND_CFG
+
+# ============================================================= #
+# Terraform automation
+# ============================================================= #
+
+write-config-tf:
+	@echo "$$TFVARS_DATA" > $(TERRAFORM_DIR)/.auto.tfvars
+	@echo "$$TF_BACKEND_CFG" > $(TERRAFORM_DIR)/backend.hcl
+
+init: write-config-tf
+	# Initializing the terraform environment
+	@terraform -chdir=$(TERRAFORM_DIR) init -input=false \
+		-reconfigure \
+		-backend-config=backend.hcl
+
+plan: init
+	# Creating all AWS infrastructure.
+	@terraform -chdir=$(TERRAFORM_DIR) plan
+
+apply: init
+	# Creating all AWS infrastructure.
+	@terraform -chdir=$(TERRAFORM_DIR) apply -auto-approve -input=false
+
+
+force-unlock: init
+	terraform -chdir=$(TERRAFORM_DIR) force-unlock $(LOCK_ID)
+
+destroy: init-tf
+	terraform -chdir=$(TERRAFORM_DIR) destroy
+
+all: build-client build-lambdas init apply
+
+# ============================================================= #
+# Builds 
+# ============================================================= #
+
+build-client:
+	rm -rf ./terraform/build/client || true
+	npm install --prefix client
+	npm run --prefix client build
+	mkdir -p ./terraform/build
+	mv ./client/build ./terraform/build/client
+
+build-lambdas:
+	
+	@echo -e "\n\n\n============================="
+	@echo "Cleaning up previous builds"
+	@echo "============================="
+	rm -rf ./terraform/build/server || true
+	mkdir -p ./terraform/build
+	
+	
+	@echo -e "\n\n\n============================="
+	@echo "Install Node Modules in all packages"
+	@echo "============================="
+	npm install --production --prefix server
+	npm install --production --prefix server/lambda/layer/common/nodejs/custom_modules
+	npm install --production --prefix server/lambda/layer/common/nodejs
+
+	@echo -e "\n\n\n============================="
+	@echo "Copy backend to temp folder to start build"
+	@echo "============================="
+	cp -r ./server ./terraform/build/server
+
+	@echo -e "\n\n\n============================="
+	@echo "Optimize node modules size"
+	@echo "============================="
+	npx modclean -n default:safe,default:caution -er -D terraform/build/server
+
+	curl -s https://gobinaries.com/tj/node-prune |  PREFIX=. sh
```

Comment:
This feels a little weird. Is this to trim down the zip file?

Reply:
Yes. This is to redue unwanted junk, so lambda zips are trim.